docs/ci/codemagic.mdx

```diff
@@ -0,0 +1,231 @@
+---
+sidebar_position: 2
+title: ♦️ Codemagic
+description: Integrate Shorebird into your Codemagic workflow
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Codemagic Workflow Integration
+
+This guide will help you integrate Shorebird into your Codemagic Workflow using the Codemagic Workflow Editor and YAML configuration.
+
+## Prerequisites
+
+✅ Shorebird CLI is installed on your machine
+
+✅ You are logged into a Shorebird account.
+
+:::info
+Refer to the [getting started](/) instructions for more information.
+:::
+
+## Authentication
+
+Most Shorebird functionality, like creating releases and patches, requires being authenticated. In order to authenticate with Shorebird in CI, you will need to generate a CI token.
+
+```sh
+shorebird login:ci
+```
+
+You will be prompted to go through a similar OAuth Flow as when using `shorebird login`, however, `shorebird login:ci` will not store any credentials on your device. Instead, a Shorebird token will be generated for you to use in CI.
+
+The output should look something like:
+
+```sh
+$ shorebird login:ci
+The Shorebird CLI needs your authorization to manage apps, releases, and patches on your behalf.
+
+In a browser, visit this URL to log in:
+
+https://accounts.google.com/o/oauth2/v2/auth...
+
+Waiting for your authorization...
+
+🎉 Success! Use the following token to login on a CI server:
+
+<SHOREBIRD_TOKEN>
+
+Example:
+
+export SHOREBIRD_TOKEN="$SHOREBIRD_TOKEN" && shorebird patch android
+```
+
+:::caution
+The `SHOREBIRD_TOKEN` is a secret and should not be committed directly in your source code or shared publicly.
+:::
+
+Next, copy the generated `SHOREBIRD_TOKEN` and navigate to your Codemagic
+secrets via:
+
+1. Go to "Environment Variables"
+2. Enter `SHOREBIRD_TOKEN` as variable name
+3. Paste the token into the variable value field
+4. Select or create new group, such as "shorebird"
+5. Ensure "Secure" is checked
+
+We recommend using the variable name `SHOREBIRD_TOKEN` as it's the default name
+used by the Shorebird CLI. If you choose a different name, you'll need to set
+the `SHOREBIRD_TOKEN` environment variable in your Codemagic workflow manually.
+
+![Screenshot of the "Environment Variables" tab in
+Codemagic](https://github.com/shorebirdtech/docs/assets/24459435/a2f1ee39-686b-4473-8f3e-d720e835824f)
+
+## Codemagic Workflow Editor
+
+_If you're using the `codemagic.yaml`, please follow the instructions in the
+[Codemagic YAML](#codemagic-yaml) section._
+
+The documentation for the Workflow Editor is currently under development. Please
+refer to the [Codemagic YAML](#codemagic-yaml) instructions in the meantime.
+
+## Codemagic YAML
+
+_If you are using Workflow Editor, please follow the instructions in [Codemagic
+Workflow Editor](#codemagic-workflow-editor) section._
+
+In the `codemagic.yaml` file, specify the `SHOREBIRD_TOKEN` under `environment`.
```

Comment:
I'm not sure how well nested tabs work, but the workflow editor/yaml sections seem like they'd be good candidates for tabs

Reply:
Today at Fluttercon I talked to @kevin-suhajda and he confirmed to me that it's not really possible to use Shorebird with the Codemagic Workflow Editor (or only in a very hacky way) because you would need to change the replace the `flutter build` command. Therefore, Kevin and I thought just to recommend using the Codemagic YAML.

Reply:
Sounds good to me! I'd just update the doc to explain this recommendation and lgtm!